### PR TITLE
Rename XRP/Models

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Rename classes in `XRP/models`. JavaScript prefers `Xrp` to `XRP`. 

### Context of Change

Deprecate the existing classes and add a new idiomatically cased classes. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

More idiomatic code is used. 

## Test Plan

CI

### CODE REVIEW

**Comment by keefertaylor** [comment]
Apologies for the bad code coverage score on this PR. 

My sense here is that since we'll remove this in N+2 releases (1 month) it isn't worth duplicating all the tests to game a coverage score. LMK if reviewers feel strongly otherwise.

File: src/XRP/model/xrp-path.ts [hunk 1]
```diff
@@ -31,3 +33,32 @@ export default class XRPPath {
...
+export default class XrpPath {
+  /**
+   * Constructs an XrpPath from a Path.
+   *
+   * @param path a Path (protobuf object) whose field values will be used
+   *             to construct an XrpPath
+   * @returns an XrpPath with its fields set via the analogous protobuf fields.
+   * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L237
+   */
+  public static from(path: Path): XRPPath {
```
Comment (amiecorso) [inline src/XRP/model/xrp-path.ts:50]: this return type should be `XrpPath` ?

File: src/XRP/model/xrp-path-element.ts [hunk 2]
```diff
@@ -43,3 +45,44 @@ export default class XRPPathElement {
...
+   *
+   * @param pathElement a PathElement (protobuf object) whose field values will be used
+   *                    to construct an XrpPathElement
+   * @returns an XrpPathElement with its fields set via the analogous protobuf fields.
+   * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L227
+   */
+  public static from(pathElement: PathElement): XrpPathElement {
+    const account = pathElement.getAccount()?.getAddress()
+    const currency = pathElement.getCurrency()
+    const xrpCurrency = currency && XRPCurrency.from(currency)
```
Comment (amiecorso) [inline src/XRP/model/xrp-path-element.ts:65]: should the idiomatically-cased classes be sure to use idiomatically-cased members?  i.e. this should be `XrpCurrency` ?

Comment (amiecorso) [reply src/XRP/model/xrp-path-element.ts:65]: I suppose we'll catch any of these stragglers when we remove the deprecated code and some things are broken

File: src/XRP/model/xrp-payment.ts [hunk 3]
```diff
@@ -103,3 +105,103 @@ export default class XRPPayment {
...
+      paymentDeliverMinValue && XrpCurrencyAmount.from(paymentDeliverMinValue)
+    if (paymentDeliverMinValue && !deliverMin) {
+      return undefined
+    }
+
+    const invoiceID = payment.getInvoiceId()?.getValue_asU8()
+
+    const paths =
+      payment.getPathsList()?.length > 0
+        ? payment.getPathsList().map((path) => XRPPath.from(path))
```
Comment (amiecorso) [inline src/XRP/model/xrp-payment.ts:163]:
```suggestion
        ? payment.getPathsList().map((path) => XrpPath.from(path))
```

File: src/XRP/model/xrp-transaction.ts [hunk 4]
```diff
@@ -208,3 +210,204 @@ export default class XRPTransaction {
...
+      case Transaction.TransactionDataCase.PAYMENT: {
+        const payment = transaction.getPayment()
+        if (!payment) {
+          return undefined
+        }
+        paymentFields = payment && XrpPayment.from(payment, xrplNetwork)
+        if (!paymentFields) {
+          return undefined
+        }
+        type = payment && XRPTransactionType.Payment
```
Comment (amiecorso) [inline src/XRP/model/xrp-transaction.ts:293]:
```suggestion
        type = payment && XrpTransactionType.Payment
```

File: src/XRP/xrp-client-decorator.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { Wallet } from 'xpring-common-js'
 import { BigInteger } from 'big-integer'
 import TransactionStatus from './transaction-status'
 import RawTransactionStatus from './raw-transaction-status'
-import XRPTransaction from './model/xrp-transaction'
+import { XRPTransaction } from './model/xrp-transaction'
```
Comment (amiecorso) [inline src/XRP/xrp-client-decorator.ts:5]: Are we just going to wait until deleting the old code to replace the references to the old class names?

File: src/XRP/model/xrp-path.ts (hunk 1 above)
Comment (keefertaylor) [reply src/XRP/model/xrp-path.ts:50]: I can't believe that compiles...

File: src/XRP/xrp-client-decorator.ts (hunk 5 above)
Comment (keefertaylor) [reply src/XRP/xrp-client-decorator.ts:5]: See #539. 

I think I have to deprecate / migrate some more XRP classes than initially intended.

**Pull request merged**